signals.c

```diff
@@ -0,0 +1,72 @@
+/*
+  Not part of Grbl. KeyMe specific.
+  
+  Signal is one layer of abstraction above adc.h and adc.c.
+
+  ADC values are read in specific time intervals, filtered and
+  stored in the appropriate arrays.
+*/
+
+#include "signals.h"
+#include "adc.h"
+#include "systick.h"
+
+#define N_FILTER 3
+#define SIGNALS_CALLBACK_INTERVAL 20
+
+uint16_t analog_voltage_readings_x[VOLTAGE_SENSOR_COUNT][N_FILTER + 1];  // Unfiltered ADC readings
+
+// Updates motors ADC readings
+void signals_update_motors()
+{
+  uint8_t idx;
+
+  for (idx = 0; idx < N_AXIS; idx++) {
+    // Assumes the motors are on ADC channels 0-3 and in the same
+    // order in analog_voltage_readings. If the pins are changed, a
+    // motors should be mapped to ADC channels.
+      analog_voltage_readings[idx] = adc_read_channel(idx);
+  }
+}
+
+// Filter and update force ADC reading
+void signals_update_force()
+{
+  #ifdef USE_LOAD_CELL
+    analog_voltage_readings_x[FORCE_VALUE_INDEX][N_FILTER] = adc_read_channel(LC_ADC);
+  #else
+    analog_voltage_readings_x[FORCE_VALUE_INDEX][N_FILTER] = adc_read_channel(F_ADC); 
+  #endif
+    /*
+    Filter:
+      Moving average Hanning Filter:
+      y[k] = 0.25 * (x[k] + 2x[k-1] + x[k-2])
+    */
+    analog_voltage_readings[FORCE_VALUE_INDEX] = (uint16_t)(
+    0.25 * (analog_voltage_readings_x[FORCE_VALUE_INDEX][N_FILTER] 
+    + (2 * analog_voltage_readings_x[FORCE_VALUE_INDEX][N_FILTER - 1]) 
+    + analog_voltage_readings_x[FORCE_VALUE_INDEX][N_FILTER - 2] ));
+
+    // Advance all values in the unfiltered array
```

Comment:
memcpy works too. I didn't think there would be a real performance benefit (I thought memcpy is implemented in a similar way) or that it is very crucial since N_FILTER = 3. But there's no harm in changing to memcpy, thanks!